File: taurus-protect-sdk-go/pkg/protect/model/business_rule.go
package model

// BusinessRule represents a business rule in the system.
type BusinessRule struct {
	// ID is the unique identifier of the business rule.
	ID string `json:"id"`
	// TenantID is the tenant identifier.
	TenantID string `json:"tenant_id,omitempty"`
	// Currency is the currency this rule applies to (if applicable).
	Currency string `json:"currency,omitempty"`
	// WalletID is the wallet ID this rule applies to (deprecated, use EntityType/EntityID).
	WalletID string `json:"wallet_id,omitempty"`
	// RuleKey is the key identifying the rule type.
	RuleKey string `json:"rule_key"`
	// RuleValue is the value of the rule.
	RuleValue string `json:"rule_value"`
	// RuleGroup is the group this rule belongs to.
	RuleGroup string `json:"rule_group,omitempty"`
	// RuleDescription is a description of the rule.
	RuleDescription string `json:"rule_description,omitempty"`
	// RuleValidation is the validation pattern for the rule value.
	RuleValidation string `json:"rule_validation,omitempty"`
	// AddressID is the address ID this rule applies to (deprecated, use EntityType/EntityID).
	AddressID string `json:"address_id,omitempty"`
	// CurrencyInfo contains detailed currency information if available.
	CurrencyInfo *Currency `json:"currency_info,omitempty"`
	// EntityType indicates what this rule applies to (global, currency, wallet, address, exchange, exchange_account, tn_participant).
	EntityType string `json:"entity_type,omitempty"`
	// EntityID is the identifier of the affected entity.
	EntityID string `json:"entity_id,omitempty"`
}

// ListBusinessRulesOptions contains options for listing business rules.
type ListBusinessRulesOptions struct {
	// IDs filters by business rule IDs.
	IDs []string
	// RuleKeys filters by rule keys.
	RuleKeys []string
	// RuleGroups filters by rule groups.
	RuleGroups []string
	// WalletIDs filters by wallet IDs (deprecated, use EntityType/EntityIDs).
	WalletIDs []string
	// CurrencyIDs filters by currency IDs.
	CurrencyIDs []string
	// AddressIDs filters by address IDs (deprecated, use EntityType/EntityIDs).
	AddressIDs []string
	// Level filters by rule level (deprecated, use EntityType).
	// One of: '', 'global', 'currency', 'address', 'wallet'.
	Level string
	// EntityType filters rules by what they apply to.
	// One of: 'global', 'currency', 'wallet', 'address', 'exchange', 'exchange_account', 'tn_participant'.
	EntityType string
	// EntityIDs filters by entity identifiers.
	EntityIDs []string
	// CurrentPage is the base64-encoded cursor for the current page.
	CurrentPage string
	// PageRequest indicates which page to request (FIRST, PREVIOUS, NEXT, LAST).
	PageRequest string
	// PageSize is the number of items per page.
	PageSize int64
}

// ListBusinessRulesResult contains the result of listing business rules.
type ListBusinessRulesResult struct {
	// BusinessRules is the list of business rules.
	BusinessRules []*BusinessRule `json:"business_rules"`
	// CurrentPage is the base64-encoded cursor for the current page.
	CurrentPage string `json:"current_page,omitempty"`
	// HasPrevious indicates if there is a previous page.
	HasPrevious bool `json:"has_previous"`
	// HasNext indicates if there is a next page.
	HasNext bool `json:"has_next"`
}

// UpdateTransactionsEnabledRequest contains the request to update the transactions enabled rule.
type UpdateTransactionsEnabledRequest struct {
	// Enabled indicates whether transactions should be enabled.
	Enabled bool `json:"enabled"`
}

File: taurus-protect-sdk-typescript/src/models/business-rule.ts
/**
 * Business rule models for Taurus-PROTECT SDK.
 *
 * Business rules define automated policies that apply to wallets, addresses,
 * or currencies. They can enforce constraints like spending limits, approval
 * requirements, or allowed transaction types.
 */

/**
 * Currency information associated with a business rule.
 */
export interface BusinessRuleCurrency {
  /** Currency ID */
  readonly id?: string;
  /** Currency symbol (e.g., ETH, BTC) */
  readonly symbol?: string;
  /** Currency name */
  readonly name?: string;
}

/**
 * Business rule model representing automated policies in Taurus-PROTECT.
 *
 * Business rules can enforce constraints like spending limits, approval requirements,
 * or allowed transaction types at various levels (global, currency, wallet, address).
 */
export interface BusinessRule {
  /** Unique identifier for the rule */
  readonly id?: string;
  /** Tenant ID the rule belongs to */
  readonly tenantId?: number;
  /** Currency ID (deprecated, use entityType/entityId) */
  readonly currency?: string;
  /** Wallet ID (deprecated, use entityType/entityId) */
  readonly walletId?: string;
  /** Address ID (deprecated, use entityType/entityId) */
  readonly addressId?: string;
  /** Rule key identifier */
  readonly ruleKey?: string;
  /** Rule value */
  readonly ruleValue?: string;
  /** Rule group for categorization */
  readonly ruleGroup?: string;
  /** Human-readable description of the rule */
  readonly ruleDescription?: string;
  /** Validation expression for the rule */
  readonly ruleValidation?: string;
  /** Currency information */
  readonly currencyInfo?: BusinessRuleCurrency;
  /** Entity type: global, currency, wallet, address, exchange, exchange_account, tn_participant */
  readonly entityType?: string;
  /** Entity ID (wallet ID, address ID, currency ID, exchange label, etc.) */
  readonly entityId?: string;
}

/**
 * Options for listing business rules.
 */
export interface ListBusinessRulesOptions {
  /** Filter by rule IDs */
  ids?: string[];
  /** Filter by rule keys */
  ruleKeys?: string[];
  /** Filter by rule groups */
  ruleGroups?: string[];
  /** Filter by wallet IDs (deprecated, use entityType/entityIds) */
  walletIds?: string[];
  /** Filter by currency IDs */
  currencyIds?: string[];
  /** Filter by address IDs (deprecated, use entityType/entityIds) */
  addressIds?: string[];
  /** Filter by level: global, currency, wallet, address (deprecated, use entityType) */
  level?: string;
  /** Filter by entity type: global, currency, wallet, address, exchange, exchange_account, tn_participant */
  entityType?: string;
  /** Filter by entity IDs */
  entityIds?: string[];
  /** Page size for pagination */
  pageSize?: number;
  /** Current page cursor for pagination */
  currentPage?: string;
  /** Page request direction: FIRST, PREVIOUS, NEXT, LAST */
  pageRequest?: 'FIRST' | 'PREVIOUS' | 'NEXT' | 'LAST';
}

/**
 * Result of listing business rules with pagination cursor.
 */
export interface ListBusinessRulesResult {
  /** List of business rules */
  readonly rules: BusinessRule[];
  /** Cursor for next page (base64-encoded) */
  readonly nextCursor?: string;
  /** Whether there are more pages */
  readonly hasMore: boolean;
}
